### DISCUSSION TITLE
is it possible to update Joomla via a CI/CD pipeline?

### DISCUSSION THREAD

**Discussion by eric-lamy-agerix** [joomla/joomla-cms#35582] [Q&A]
Hello,
Do you have experience with Joomla sites updated via a CI/CD pipeline? I am looking for solutions to update the DB and the code via an automated CI/CD pipeline (like Git) because the client's IT infrastructure refuses to update via installable packages, as these do not allow rollback in case of problems.
Thank you in advance for your answers

**Comment by joomlaweby** [comment]
Try to check Joomla CLI - [https://magazine.joomla.org/all-issues/june-2022/joomla-4-a-powerful-cli-application](https://magazine.joomla.org/all-issues/june-2022/joomla-4-a-powerful-cli-application)

There is a cheat sheet with CLI commands in this article.

**Comment by noxidsoft** [comment]
@agerix hmm, have you had any other discussion, with any other interested party, around this since you asked the question? I'm curious too, about the possibilities. It might be something we need to create, at least experiment with. It might make a good community side-project.

**Comment by noxidsoft** [comment]
Because a top Google search brought me to this discussion first, I'll include these known resources here, hope someone finds them useful:

- https://docs.joomla.org/Archived:How_to_create_a_continuous_integration (circa. 2013)
- https://www.youtube.com/watch?v=H5JgIQTLoO0 (circa. 2017)
- https://techjoomla.com/developers-blogs/packaging-system-for-joomla-extensions (circa. 2020)
- https://www.slideshare.net/winggundamth/joomla-continuous-delivery-with-docker (circa. 2014)

https://www.jenkins.io/ (OpenSource)

**Comment by Llewellynvdm** [comment]
@agerix we are looking at this https://github.com/joomla-docker/docker-joomla/issues/161 and it is indeed something we would like make part of the official Joomla images.
